StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedFormViewController.swift

```diff
@@ -0,0 +1,430 @@
+            if error != nil {
+                // TODO(porter) Don't think we need this
+                // Without this, setting `contentOffsetPercentage = 1` uses the height of the scroll view without accounting for the error label
+                errorLabel.setNeedsLayout()
+                errorLabel.layoutIfNeeded()
+            }
+        }, postLayoutAnimations: {
+            if self.error != nil {
+                // Scroll the view to the bottom to ensure the error is visible
+                self.bottomSheetController?.contentOffsetPercentage = 1
+            }
+        })
+    }
+
+    private func didCancel() {
+        delegate?.embeddedFormViewControllerDidCancel(self)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - UIViewController Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = configuration.appearance.colors.background
+        configuration.style.configure(self)
+        paymentContainerView.directionalLayoutMargins = .zero
+
+        // One stack view contains all our subviews
+        let views: [UIView] = [paymentContainerView, mandateView, errorLabel].compactMap { $0 }
+        for view in views {
+            stackView.addArrangedSubview(view)
+        }
+        stackView.spacing = 20
+        stackView.directionalLayoutMargins = PaymentSheetUI.defaultMargins
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.axis = .vertical
+        stackView.sendSubviewToBack(mandateView)
+
+        for subview in [stackView, primaryButton] {
+            subview.translatesAutoresizingMaskIntoConstraints = false
+            view.addSubview(subview)
+        }
+        NSLayoutConstraint.activate([
+            stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            primaryButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: PaymentSheetUI.defaultSheetMargins.leading),
+            primaryButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -PaymentSheetUI.defaultSheetMargins.trailing),
+
+            stackView.topAnchor.constraint(equalTo: view.topAnchor),
+            stackView.bottomAnchor.constraint(equalTo: primaryButton.topAnchor, constant: -32),
+            primaryButton.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -PaymentSheetUI.defaultSheetMargins.bottom),
+        ])
+    }
+
+    private var didSendLogShow: Bool = false
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if !didSendLogShow {
+            // Only send this once to match the behavior of horizontal mode
+            didSendLogShow = true
+            analyticsHelper.logShow(showingSavedPMList: false) // We never show the saved PM list first
+        }
```

Comment:
Per https://docs.google.com/document/d/1hBpsyWQi2d_gGMkS-_aUGUZVn1g53mjlLGytyxraqnU/edit?tab=t.0#heading=h.gnypjvjc9wbz we should delete this; we don't call `logShow`.

Reply:
Removing!